node/src/components/storage/tests.rs

```diff
@@ -1152,3 +1152,90 @@ fn should_hard_reset() {
     // Test with a hard reset to era 0, deleting all blocks and associated data.
     check(0);
 }
+
+#[test]
+fn can_put_and_get_blocks_v2() {
+    let num_blocks = 10;
+    let mut harness = ComponentHarness::default();
+    let mut storage = storage_fixture(&harness);
+
+    let era_id = harness.rng.gen_range(0..10).into();
+    let height = harness.rng.gen_range(0..100);
+
+    let mut blocks = vec![];
+
+    for i in 0..num_blocks {
+        let block = Block::random_with_specifics(
+            &mut harness.rng,
+            era_id,
+            height + i,
+            HashingAlgorithmVersion::HASH_V2_PROTOCOL_VERSION,
+            i == num_blocks - 1,
+        );
+
+        blocks.push(block.clone());
+
+        assert!(put_block(
+            &mut harness,
+            &mut storage,
+            Box::new(block.clone())
+        ));
+
+        let mut txn = storage.env.begin_ro_txn().unwrap();
+        let block_body_merkle = block.body().merklize();
+
+        for hash in [
+            *block_body_merkle
+                .deploy_hashes
+                .merkle_linked_list_node_hash(),
+            *block_body_merkle
+                .transfer_hashes
+                .merkle_linked_list_node_hash(),
+            *block_body_merkle.proposer.merkle_linked_list_node_hash(),
+        ] {
```

Comment:
You mentioned that we could use the `take_hashes_and_proofs` function I whipped up [here][1].  We could then make that function `#[cfg(test)]` rather than `pub`...

Maybe we can just roll these two PRs together?

[1]: https://github.com/casper-network/casper-node/pull/1708/files#diff-f8f1402a94ae96f1e79dfff1d2af53d86a2114c73096a211873dbbb4fe4226e9R1037-R1054